### ISSUE TITLE
Add Inter-Region VPC Peering Connection Support

### ISSUE THREAD

**Issue by bflad** [terraform-providers/terraform-provider-aws#2484]
AWS has announced support for VPC peering across regions: https://aws.amazon.com/about-aws/whats-new/2017/11/announcing-support-for-inter-region-vpc-peering/

Prerequisite: aws-sdk-go v1.12.38 (#2500)

### Terraform Version

```
terraform 0.10+
terraform-provider-aws 1.5.0
```

### Affected Resource(s)

- `aws_vpc_peering_connection`

### Expected Behavior

Optional `peer_region` attribute available and utilized.

```hcl
resource "aws_vpc_peering_connection" "foo" {
  peer_owner_id = "${var.peer_owner_id}"
  peer_region   = "us-east-1"
  peer_vpc_id   = "${aws_vpc.bar.id}"
  vpc_id        = "${aws_vpc.foo.id}"
}
```

### References

http://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_CreateVpcPeeringConnection.html

**Comment by bflad** [comment]
AWS SDK functionality is in v1.12.38, which is has a pull request now: #2500

**Comment by jen20** [comment]
This is fixed by #2508.

**Comment by bflad** [comment]
This has been released in [terraform-provider-aws version 1.7.0](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#170-january-12-2018). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.
